### PR TITLE
GTiff and COG: add a MAX_Z_ERROR_OVERVIEW creation option …

### PR DESCRIPTION
**Pull request by rouault**
…to separately control the error threshold of overviews w.r.t the one of the full resolution image

### CODE REVIEW

**Comment by blacha** [comment]
TLDR: This change looks great, thanks @rouault :tada: !

I have been doing a little bit of experimentation with this change, most of our datasets have a vertical error in the 5-20cm range and we store them as "close enough" to lossless with 0.001 error (1mm)

Below is a table comparing the size differences for a random tile when changing the 
`MAX_Z_ERROR_OVERVIEW` value, 

With this change and 0.1 (10cm) z error would give us a ~15% file size reduction while still preserving the root image as near lossless :tada: 


|file_name                                                   |overview_error|file_size_mb|file_size_vs_source_mb|file_size_diff_percent|
|------------------------------------------------------------|--------------|------------|-------------------|----------------------|
|./output/cf15_comp-lerc_z-error-0.001.tiff                  |0.001         |1319.106134 |                   |                      |
|./output/cf15_comp-lerc_z-error-0.001_z-ovr-error-0.002.tiff|0.002         |1288.947152 |-30.15898228       |-2.29%                |
|./output/cf15_comp-lerc_z-error-0.001_z-ovr-error-0.005.tiff|0.005         |1249.09688  |-70.00925446       |-5.31%                |
|./output/cf15_comp-lerc_z-error-0.001_z-ovr-error-0.01.tiff |0.01          |1219.147237 |-99.95889759       |-7.58%                |
|./output/cf15_comp-lerc_z-error-0.001_z-ovr-error-0.02.tiff |0.02          |1189.420308 |-129.6858263       |-9.83%                |
|./output/cf15_comp-lerc_z-error-0.001_z-ovr-error-0.05.tiff |0.05          |1150.768214 |-168.3379202       |-12.76%               |
|./output/cf15_comp-lerc_z-error-0.001_z-ovr-error-0.1.tiff  |0.1           |1122.241355 |-196.8647795       |-14.92%               |
